sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,147 @@
+    b = (p1.x - p1.z)**2%n
+    An, Ad = A
+    ab = a - b
+    x_cor = 4*a*b*Ad % n
+    z_cor = (4*Ad*b + ab*An)*ab % n
+    return point(x_cor%n, z_cor%n)
+
+def _mont_ladder(p_0, k, A, n):
+    """Scalar multiplication of points done
+    using Montgomery Ladder algorithm.
+
+    Parameters
+    ==========
+    p_0 : point on the curve
+    k : Integer
+    A : parameter of the curve
+    n : Modulus
+    """
+    k = bin(k)[3:]
+    q = p_0
+    p = _pt_double(p_0, A, n)
+    for i in k:
+        if i == '1':
+            q = _pt_add(p, q, p_0, n)
+            p = _pt_double(p, A, n)
+        else:
+            p = _pt_add(q, p, p_0, n)
+            q = _pt_double(q, A, n)
+    return q
+
+def ilog(x, b):
+    """Returns least integer l for which b**l >= x"""
+    l = 0
+    while x >= b:
+        x /= b
+        l += 1
+    return l
+
+
+def ecm(n, B1=100, B2=200, max_curve=5):
+    """Lenstra's Elliptic curve Factorization using
+    2 stage with Suyuma's Parameterization.
+
+    References
+    ==========
+
+    .. [1]  Carl Pomerance and Richard Crandall "Prime Numbers:
+        A Computational Perspective" (2nd Ed.), page 344
+    """
+    from sympy.ntheory import prime, sieve, isprime
+    from sympy import gcd
+    import random
+    factors = set()
+    # Find all the small primefactors
+    for i in sieve.primerange(1, 10000):
+        if n % i == 0:
+            factors.add(i)
+            while(n % i == 0):
+                n = n//i
+                n = int(n)
+    curve = 0
+    while n > 1:
+        curve += 1
+        #Max curve reached for the bound, time to increase the bounds
+        if curve > max_curve:
+            max_curve *= 3
+            curve = 0
+            B1 *= 2
+            B2 *= 2
+        if isprime(n):
+            factors.add(n)
+            break
+        i  = 1
+        #Suyuma Parametrization
+        sigma = random.randint(6, n - 1)
+        u = (sigma**2 - 5) % n
+        v = 4*sigma % n
+        p = pow(u, 3, n)
+        Q = point(pow(v - u, 3, n)*(3*u + v) % n, 4*p*v % n)
+        C = (p, pow(v, 3, n))
+        p = prime(i)
+        i += 1
+        #Stage 1
+        while p <= B1:
+            Q = _mont_ladder(Q, p**ilog(B1, p), C, n)
+            p = prime(i)
+            i += 1
+        g = gcd(Q.z, n)
+        if 1 < g < n:
+            factors.add(g)
+            n = n//g
+            n = int(n)
+            B1 = 100
+            B2 = 200
+        if isprime(n):
+            factors.add(n)
+            break
+        while p <= B2:
+            Q = _mont_ladder(Q, p, C, n)
+            g *= Q.z
+            g %= n
+            p = prime(i)
+            i += 1
+        g = gcd(g, n)
+        if 1 < g < n:
+            factors.add(g)
+            n //= g
+            n = int(n)
+            B1 = 100
+            B2 = 200
+    final = set()
+    for i in factors:
+        if isprime(i):
+            final.add(i)
+        else:
+            final |= ecm(i)
+    return set(sorted(final))
```

Comment:
No need to sort if you are putting into a set.

Reply:
On the otherhand, if you are mimicking `primefactors` then `sorted(set(final))` would be appropriate. But since `final` is already a set just `sorted(final)` is all that is needed.